File: 2021/day05/solution2.pl
#! /usr/bin/perl

while (<>) { @xy = m/\d+/g; do { $m{$z = "@xy[0,1]"}++; $xy[$_] += $xy[$_+2] <=> $xy[$_] for (0, 1); } while $z ne "@xy[2,3]" }
my $res = scalar grep { $_>1 } values %m;
printf "🎄2: %d\n", $res;

File: 2021/day08/solution2.jl
#!/usr/bin/env julia
using Printf
i = split.(readlines("input.txt"),'|')
# following ideas from /r/adventofcode/comments/rbj87a/comment/hnp4saz
res = sum(parse.(BigInt,["4725360918"[[sum([X...].∈e)÷2%15%11+1 for e in split(Y)]] for (X,Y) in i]))
@printf "🎄2: %d \n" res